windows/win_robocopy.ps1

```diff
@@ -0,0 +1,130 @@
+#!powershell
+# This file is part of Ansible
+#
+# Copyright 2015, Corwin Brown <corwin.brown@maxpoint.com>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$params = Parse-Args $args;
+
+$result = New-Object psobject @{
+    win_robocopy = New-Object psobject @{
+        recurse         = $false
+        purge           = $false
+    }
+    changed = $false
+}
+
+$src = Get-AnsibleParam -obj $params -name "src" -failifempty $true
+$dest = Get-AnsibleParam -obj $params -name "dest" -failifempty $true
+$purge = Get-AnsibleParam -obj $params -name "purge" -default $false
+$recurse = Get-AnsibleParam -obj $params -name "recurse" -default $false
+$flags = Get-AnsibleParam -obj $params -name "flags" -default $null
+
+# Search for an Error Message
+# Robocopy seems to display an error after 3 '-----' separator lines
+Function SearchForError($cmd_output, $default_msg) {
+    $separator_count = 0
+    $error_msg = $default_msg
+    ForEach ($line in $cmd_output) {
+        if (-Not $line) {
+            continue
+        }
+
+        if ($separator_count -ne 3) {
+            if (Select-String -InputObject $line -pattern "^(\s+)?(\-+)(\s+)?$") {
+                $separator_count += 1
+            }
+        }
+        Else {
+            If (Select-String -InputObject $line -pattern "error") {
+                $error_msg = $line
+                break
+            }
+        }
+    }
+
+    return $error_msg
+}
+
+# Build Arguments
+$robocopy_opts = @()
+
+if (-Not (Test-Path $src)) {
+    Fail-Json $result "$src does not exist!"
+}
+
+$robocopy_opts += $src
+Set-Attr $result.win_robocopy "src" $src
+
+$robocopy_opts += $dest
+Set-Attr $result.win_robocopy "dest" $dest
+
+if ($flags -eq $null) {
+    if ($purge) {
+        $robocopy_opts += "/purge"
+    }
+
+    if ($recurse) {
+        $robocopy_opts += "/e"
+    }
+}
+Else {
+    $robocopy_opts += $flags
+}
+
+Set-Attr $result.win_robocopy "purge" $purge
+Set-Attr $result.win_robocopy "recurse" $recurse
+Set-Attr $result.win_robocopy "flags" $flags
+
+Try {
```

Comment:
Hi blakfeld - thanks again for the PR.  It would be nice to have support for `check_mode` here ala https://github.com/ansible/ansible-modules-extras/blob/devel/windows/win_updates.ps1#L145